types/prettier/index.d.ts

```diff
@@ -256,7 +256,7 @@ export function clearConfigCache(): void;
 export interface SupportLanguage {
     name: string;
     since?: string;
-    parsers: string[];
+    parsers: BuiltInParserName[];
```

Comment:
I think it'd better to leave it as-is since this can be used by external [plugins](https://prettier.io/docs/en/plugins.html).

Reply:
I have been using it to develop a vscode plugin and that gave me a headache

<img width="863" alt="screenshot 2018-11-24 at 3 45 46 pm" src="https://user-images.githubusercontent.com/20282546/48966980-1b17a200-f000-11e8-8019-db9ee67eb870.png">


It returns a string which is incompatiable with type builtinparsername

Reply:
@ikatyang I first added `customParser` then I removed it. https://github.com/DefinitelyTyped/DefinitelyTyped/compare/2090f122a55e1aaf34bdffc94d5d4eef16927376..e0a0256b1706edcbe9079d492c34e28f3682ad4a.

Should I revert that back. It should be compatible with external plugin then?

Reply:
Or something like `BuiltInParser[] | CustomParser[] | string[]` so that it will be compatible either ways.

Reply:
Thanks @ikatyang . That solves too. But would be nice though if available by default.

Reply:
`CustomParser` will never be returned by `.getSupportInfo()` since it's only allowed to be used in API/CLI, there's no `CustomParser` for plugins and `.getSupportInfo()` only works for plugins (including built-in ones).

And `BuiltInParser[] | string[]` will be merged into `string[]` by TS as far as I can tell.

Reply:
`BuiltInParser[] | string[]` at-least won't throw error on strict mode

Reply:
You're right. I forgot TS won't merge union of array.